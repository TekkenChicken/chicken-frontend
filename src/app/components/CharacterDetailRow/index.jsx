import React, { Component } from "react";

export default class CharacterDetailRow extends Component {
    constructor(props) {
        super(props);
        this.state = {
            editable: false
        };

        this.updateRow = this.updateRow.bind(this);
    }

    updateRow(e) {
        console.log('clicked', e.currentTarget);
        this.setState({editable: true});
    }

    render() {
        if(this.state.editable) {
            return (
                <tr onClick={(e) => this.updateRow(e)}>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.name}/>
                        </div>
                    </td>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.notation}/>
                        </div>
                    </td>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.hit_level}/>
                        </div>
                    </td>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.damage}/>
                        </div>
                    </td>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.speed}/>
                        </div>
                    </td>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.on_block}/>
                        </div>
                    </td>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.on_hit}/>
                        </div>
                    </td>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.on_ch}/>
                        </div>
                    </td>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.properties}/>
                        </div>
                    </td>
                    <td>
                        <div className="control">
                            <input className="input" value={this.props.data.notes === "null" ? "" : this.props.data.notes}/>
                        </div>
                    </td>
                </tr>
            );
        } else {
            return (
                <tr onClick={(e) => this.updateRow(e)}>
                    <td>{this.props.data.name}</td>
                    <td>{this.props.data.notation}</td>
                    <td>{this.props.data.hit_level}</td>
                    <td>{this.props.data.damage}</td>
                    <td>{this.props.data.speed}</td>
                    <td>{this.props.data.on_block}</td>
                    <td>{this.props.data.on_hit}</td>
                    <td>{this.props.data.on_ch}</td>
                    <td>{this.props.data.properties}</td>
                    <td>{this.props.data.notes === "null" ? "" : this.props.data.notes}</td>
                </tr>
            );
        }
    }
}
