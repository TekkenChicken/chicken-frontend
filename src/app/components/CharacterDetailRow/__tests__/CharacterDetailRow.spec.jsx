import React from "react";
import { mount } from "enzyme";
import * as renderer from "react-test-renderer";

import stubFrameData from '../../../fixtures/framedata.json';
import CharacterDetailRow from "../";

describe("CharacterDetailRow", () => {
    it("should render as expected", () => {
        //select a character from the stub data
        const character = stubFrameData["king"];
        const data = character.data[2];
        const component = renderer.create(<CharacterDetailRow data={data}/>).toJSON();
        expect(component).toMatchSnapshot();
    })

    it("should allow editing onClick", () => {
        //select a character from the stub data
        const character = stubFrameData["king"];
        const data = character.data[2];
        const component = mount(<CharacterDetailRow data={data}/>);

        component.simulate('click');
        const control = component.find('.control input');
        expect(control.exists()).toBe(true);
    })
});
