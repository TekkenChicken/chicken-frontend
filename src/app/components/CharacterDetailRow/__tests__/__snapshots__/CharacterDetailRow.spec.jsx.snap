// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetailRow should render as expected 1`] = `
<tr
  onClick={[Function]}
>
  <td />
  <td>
    1, 2
  </td>
  <td>
    h, h
  </td>
  <td>
    9,20
  </td>
  <td>
    10
  </td>
  <td>
    -1
  </td>
  <td>
    +6
  </td>
  <td>
    +6
  </td>
  <td />
  <td>
    
  </td>
</tr>
`;
